### PR TITLE
[MAINTENANCE] Update default candidate strings SimpleDateFormatString parameter builder

### PR DESCRIPTION
**Pull request by NathanFarmer**
Changes proposed in this pull request:
- Follow-up PR for GREAT-493
- Add additional default candidate strings for SimpleDateFormatStringParameterBuilder
- Remove argument `*` pattern from all `ParameterBuilder`s
- Leverage `get_parameter_value_and_validate_return_type` for threshold

### Definition of Done
- [X] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [X] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [X] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py
Comment (NathanFarmer) [inline great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py]: Get threshold using utility function

File: great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py
Comment (alexsherstinsky) [inline great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py]: @NathanFarmer Type hint, please.  Thanks!

File: great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py
Comment (alexsherstinsky) [inline great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py:227]: @NathanFarmer Type hint for fmt_string, ratio please.  Thanks.

**Review by alexsherstinsky — CHANGES_REQUESTED** [review]
Small things -- otherwise, fine.

File: great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py
Comment (alexsherstinsky) [inline great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py:122]: In this code, we have tuple instead of dictionary -- extra parentheses must be removed.
```
                match_strftime_metric_value_kwargs: dict = (
                    {
                        **self._metric_value_kwargs,
                        **{"strftime_format": fmt_string},
                    },
                )
```
@NathanFarmer -- thanks.  Also, stylistically, comma after each dictionary key/value pair, even if the only one.  Thanks!

Comment (NathanFarmer) [reply great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py:122]: ✅

File: great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py
Comment (NathanFarmer) [reply great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py:227]: ✅

File: great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py
Comment (NathanFarmer) [reply great_expectations/rule_based_profiler/parameter_builder/simple_date_format_string_parameter_builder.py]: ✅

**Review by cdkini — COMMENTED** [review]
LGTM but will defer to someone a bit more knowledgeable about this candidate list matter.

Happy to approve if need be just let me know!

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM to me as well :) Thank you for establishing the pattern that can effectively be used for additional ParameterBuilders

**Review by alexsherstinsky — APPROVED** [review]
LGTM!

**Pull request merged**
